src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/subscribe_button_request_test.cc

```diff
@@ -0,0 +1,141 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "hmi/subscribe_button_request.h"
+
+#include <memory>
+#include <string>
+
+#include "gtest/gtest.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace hmi_commands_test {
+namespace subscribe_button_request {
+
+using ::testing::_;
+using ::testing::Return;
+namespace am = ::application_manager;
+namespace strings = am::strings;
+using am::commands::CommandImpl;
+using am::commands::RequestToHMI;
+using am::event_engine::Event;
+using sdl_rpc_plugin::commands::hmi::SubscribeButtonRequest;
+using ::test::components::application_manager_test::MockApplication;
+
+typedef std::shared_ptr<MockApplication> MockAppPtr;
+typedef std::shared_ptr<RequestToHMI> RequestToHMIPtr;
+typedef std::shared_ptr<SubscribeButtonRequest> SubscribeButtonRequestPtr;
+
+namespace {
+const uint32_t kCorrelationId = 2u;
+const uint32_t kAppId = 1u;
+const hmi_apis::Common_ButtonName::eType kButtonName =
+    hmi_apis::Common_ButtonName::CUSTOM_BUTTON;
+const hmi_apis::FunctionID::eType kFunctionID =
+    hmi_apis::FunctionID::Buttons_SubscribeButton;
+}  // namespace
+
+class HMISubscribeButtonRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ protected:
+  MessageSharedPtr CreateCommandMsg() {
+    MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+    (*command_msg)[strings::msg_params][strings::app_id] = kAppId;
+    (*command_msg)[strings::msg_params][strings::button_name] = kButtonName;
+    (*command_msg)[strings::params][strings::correlation_id] = kCorrelationId;
+    (*command_msg)[strings::params][strings::function_id] = kFunctionID;
+
+    return command_msg;
+  }
+
+  void InitCommand(const uint32_t& timeout) OVERRIDE {
+    mock_app = CreateMockApp();
+    CommandRequestTest<CommandsTestMocks::kIsNice>::InitCommand(timeout);
+    ON_CALL((*mock_app), hmi_app_id()).WillByDefault(Return(kAppId));
+    ON_CALL(app_mngr_, application_by_hmi_app(kAppId))
+        .WillByDefault(Return(mock_app));
+    ON_CALL(app_mngr_, application(kAppId)).WillByDefault(Return(mock_app));
+  }
+
+  MockAppPtr mock_app;
+};
+
+TEST_F(HMISubscribeButtonRequestTest, Run_SendRequest) {
+  MessageSharedPtr command_msg = CreateCommandMsg();
+  RequestToHMIPtr command(CreateCommand<SubscribeButtonRequest>(command_msg));
+
+  EXPECT_CALL(mock_rpc_service_, SendMessageToHMI(command_msg));
+  ASSERT_TRUE(command->Init());
+
+  command->Run();
```

Comment:
@OlhaVorobiova I believe you can split this unit tests by two separate tests - one with `CUSTOM_BUTTON` subscription and another one with `SEEK_LEFT`. Then in the first unit test you may check
```
MockEventDispatcher mock_event_dispatcher;
  EXPECT_CALL(app_mngr_, event_dispatcher())
      .WillOnce(ReturnRef(mock_event_dispatcher));
  EXPECT_CALL(mock_event_dispatcher, add_mobile_observer(kFunctionID, kCorrelationId));
```
And in the second one
```
  EXPECT_CALL(app_mngr_, event_dispatcher())
      .WillOnce(ReturnRef(mock_event_dispatcher));
  EXPECT_CALL(mock_event_dispatcher, add_mobile_observer(kFunctionID, kCorrelationId))
     .Times(0);
```